### PR TITLE
Glasgow | 26-ITP-May | Sandani Kannangara | Sprint 3 | Implement And Rewrite Tests

### PR DESCRIPTION
**Pull request by SandzSoft**
Self checklist

- [X] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [X] My changes meet the requirements of the task
- [X] I have tested my changes
- [X] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)

## Changelist

Completed the "1-implement-and-rewrite-tests" section of Sprint 3.

- Implemented getAngleType
- Implemented isProperFraction
- Implemented getCardValue
- Added assertion tests in the implement files
- Rewrote tests using Jest
- Covered valid, boundary, and invalid test cases

### CODE REVIEW

File: Sprint-3/1-implement-and-rewrite-tests/implement/3-get-card-value.js [hunk 1]
```diff
@@ -22,7 +22,14 @@
...
 
 function getCardValue(card) {
-  // TODO: Implement this function
+  let rank = card.slice(0, -1);
+  let suit = card.slice(-1);
+  const suitArray = ["♠", "♥", "♦", "♣"];
+  if (!suitArray.includes(suit)) throw new Error("Invalid card");
+  if (rank === "A") return 11;
+  if (rank === "Q" || rank === "K" || rank === "J") return 10;
+  if (Number(rank) >= 2 && Number(rank) <= 10) return Number(rank);
```
Comment (cjyuan) [inline Sprint-3/1-implement-and-rewrite-tests/implement/3-get-card-value.js]: In JavaScript, strings that represent valid numeric literals in the language can be safely converted to equivalent numbers. For examples, "0x02", "2.1", or "0002".

Does your function return the value you expected from each of the following function calls?

```
getCardValue("0x02♠");
getCardValue("2.1♠");
getCardValue("0002♠");
```

File: Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/2-is-proper-fraction.test.js [hunk 2]
```diff
@@ -4,7 +4,34 @@ const isProperFraction = require("../implement/2-is-proper-fraction");
...
+  });
+
+  test("handles improper negative fractions", () => {
+    expect(isProperFraction(-7, 4)).toEqual(false);
+    expect(isProperFraction(7, -4)).toEqual(false);
+  });
+
+  test("returns false when numerator equals denominator", () => {
+    expect(isProperFraction(4, 4)).toEqual(false);
+  });
```
Comment (cjyuan) [inline Sprint-3/1-implement-and-rewrite-tests/rewrite-tests-with-jest/2-is-proper-fraction.test.js:40]: Could also test cases with negative numerator and negative denominator.

File: Sprint-3/1-implement-and-rewrite-tests/implement/3-get-card-value.js (hunk 1 above)
Comment (SandzSoft) [reply Sprint-3/1-implement-and-rewrite-tests/implement/3-get-card-value.js]: Thank you for your review and feedback. I revisited my solution and realised that using arrays is a better approach for this task. It makes the validation logic much simpler and more readable than handling everything with string conversions and regular expressions. I've updated my code accordingly.

**Comment by cjyuan** [comment]
Changes look good.

**Comment by illicitonion** [comment]
Closing PR because the May ITP run has finished. Feel free to re-open if you're still working on it.
